crates/http/src/error.rs

```diff
@@ -1,14 +1,15 @@
-use std::fmt::{Debug, Display};
+use std::fmt::Debug;
 
 use crate::service::{HttpService, HttpServiceFactory};
 
 /// An error that can occur when creating or running an HTTP server.
 #[derive(Debug, thiserror::Error)]
-pub enum Error<S>
+pub enum Error<F>
 where
-    S: HttpServiceFactory,
-    S::Error: Debug + Display,
-    <S::Service as HttpService>::Error: Debug + Display,
+    F: HttpServiceFactory,
+    F::Error: Debug,
+    <F::Service as HttpService>::Error: Debug,
+    <<F::Service as HttpService>::ResBody as http_body::Body>::Error: Debug,
```

Comment:
Is there any harm in requiring that this error be a `std::error::Error`? If not we should update it everywhere.